src/sentry/api/invite_helper.py

```diff
@@ -170,19 +183,36 @@ def accept_invite(self, user=None):
         if self.member_already_exists:
             self.handle_member_already_exists()
             om.delete()
-        else:
-            om.set_user(user)
-            om.save()
-
-            create_audit_entry(
-                self.request,
-                actor=user,
-                organization=om.organization,
-                target_object=om.id,
-                target_user=user,
-                event=AuditLogEntryEvent.MEMBER_ACCEPT,
-                data=om.get_audit_log_data(),
-            )
+            return
+
+        try:
+            provider = AuthProvider.objects.get(organization=om.organization)
+        except AuthProvider.DoesNotExist:
+            provider = None
+
+        # If SSO is required, check for valid AuthIdentity
+        if provider and not provider.flags.allow_unlinked:
+            try:
+                authId = AuthIdentity.objects.get(user=user)
+            except AuthIdentity.DoesNotExist:
+                authId = None
+
+            if not authId or not authId.auth_provider_id == provider.id:
+                self.handle_member_has_no_sso()
```

Comment:
Should we be raising an error here so that we can return something useful to the frontend?

Reply:
I removed the erroneous frontend elements that lead into this code path, so AFAIK it should not happen. This check inside `accept_invite` is an additional/redundant guard. 

The frontend should lead them back to the "Login with SSO" screen, where there's only 1 button and they can do the right thing from there.